sycl/doc/PreprocessorMacros.md

```diff
@@ -38,16 +38,17 @@ This file describes macros that have effect on SYCL compiler and run-time.
 
   Disables a message which warns about unsupported C++ version.
 
-- **SYCL_DISABLE_FALLBACK_ASSERT**
-
-  Defining this macro eliminates some overhead that is associated with
-  submitting kernels that call `assert()`. When this macro is defined, the logic
-  for detecting assertion failures in kernels is disabled, so a failed assert
-  will not cause a message to be printed and will not cause the program to
-  abort. However, this macro only affects kernels that are submitted to devices
-  that do **not** have native support for `assert()` because devices with native
-  support do not impose any extra overhead. One can check to see if a device has
-  native support for `assert()` via `aspect::ext_oneapi_native_assert`.
+- **SYCL_ENABLE_FALLBACK_ASSERT**
+
+  Defining this macro enables fallback assert feature. Be aware that this will
+  add some overhead that is associated with submitting kernels that call
+  `assert()`. When this macro is not defined, the logic for detecting assertion
+  failures in kernels is disabled, so a failed assert will not cause a message
+  to be printed and will not cause the program to abort. However, this macro
+  only affects kernels that are submitted to devices that do **not** have native
+  support for `assert()` because devices with native support do not impose any
+  extra overhead. One can check to see if a device has native support for
```

Comment:
I would change this sentence:

> However, this macro only affects kernels that are submitted to devices that do **not** have native support for `assert()` because devices with native support do not impose any extra overhead.

To something like:

> Some devices have native support for assertions.  The logic for detecting assertion failures is always enabled on these devices regardless of whether this macro is defined because that logic does not add any extra overhead.  One can check ...

Reply:
Fixed.